src/homework06.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# coding: utf8
+
+"""
+EPAM python q4 homework 06
+Sergey Streltsov 2019-11-17
+Simplest object of a Room
+"""
+
+import hashlib
+
+
+class Room(object):
+
+    def __init__(self, room_number, bed_number, is_lux=False):
+        self._room_number = room_number
+        self._bed_number = bed_number
+        self._is_lux = is_lux
+        self._booking_history = dict()
+
+    def book_room(self, checkin_date, checkout_date):
+        if self.check_booking_interval(
+                checkin_date=checkin_date,
+                checkout_date=checkout_date) and not self.is_booked():
+            confirm_obj = hashlib.md5((checkin_date + checkout_date).encode())
+            confirmation_number = confirm_obj.hexdigest()
+            print('Booking Ok, checkin: {}, checkout: {}'.format(
+                checkin_date, checkout_date))
+            print('Conf. number: {}'.format(confirmation_number))
+            self._booking_history[confirmation_number] = [checkin_date,
+                                                          checkout_date]
+        else:
+            print('Error: {}-{} not available'.format(checkin_date,
+                                                      checkout_date))
+            return -1
+        return confirmation_number
+
+    def cancel_booking(self, confirmation_number):
+        if confirmation_number in self._booking_history.keys():
+            dates = self._booking_history.pop(confirmation_number)
+            print('{}, booking canceled'.format(dates))
+        else:
+            print('Error: booking with conf. number {} not found'.format(
+                confirmation_number))
+
+    def check_booking_interval(self, checkin_date, checkout_date):
+        # check is interval valid
+        return bool(checkin_date and checkout_date)
+
+    def is_booked(self, mock_obj=False):
+        # check is booked
+        return mock_obj
```

Comment:
it's unclear for me. more of that we usually use mocks for tests

Reply:
yes, but i don't have time for implement logic for is_booking check, it's always return False, i told you about it, so i can rename this variable )